File: Springboot Code/AgroRent/src/main/java/com/majorproj/agrorent/entities/Booking.java
 package com.majorproj.agrorent.entities;

import java.time.LocalDate;
import java.util.List;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "Booking")
@ToString(callSuper = true,exclude = {"payment","farmer","equipment"})
public class Booking extends BaseEntity{
	@Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDate startDate;
    private LocalDate endDate;

    @Enumerated(EnumType.STRING)
    private BookingStatus status; 
    @Column(nullable = false)
    private double totalAmount;

    @ManyToOne
    @JoinColumn(name = "equipment_id")
    private Equipment equipment;

    @ManyToOne
    @JoinColumn(name = "renter_farmer ")
    private Farmer farmer;

    @OneToOne(mappedBy = "booking")
    private Payment payment;
    
    
}

File: Dot Net Code/AgroRent/Models/Booking.cs
using System.ComponentModel.DataAnnotations;

namespace AgroRent.Models
{
    public class Booking : BaseEntity
    {
        [Key]
        public int Id { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public BookingStatus Status { get; set; }

        [Required]
        public double TotalAmount { get; set; }

        // Navigation properties
        public virtual Equipment? Equipment { get; set; }

        public virtual Farmer? Farmer { get; set; }

        public virtual Payment? Payment { get; set; }
    }
}
